api/src/utils/generate-favicon.ts

```diff
@@ -17,18 +17,8 @@ const svg = (color: string, addDirectusLogo: boolean) => `
 	}
 </svg>`;
 
-export const generateFavicon = (color: string, addDirectusLogo = true) => {
-	color = color || cssVar('--theme--primary');
-
-	const icon = svg(color, addDirectusLogo);
-	const wrapper = document.createElement('div');
-	wrapper.innerHTML = icon.trim();
-
-	if (wrapper.firstChild) {
-		const iconSerialized = new XMLSerializer().serializeToString(wrapper.firstChild);
-
-		return 'data:image/svg+xml;base64,' + window.btoa(iconSerialized);
-	}
-
-	throw Error(`Couldn't generate favicon`);
+export const generateFavicon = async (color: string, addDirectusLogo = true) => {
+	const resvg = new Resvg(svg(color, addDirectusLogo));
+	const buffer = resvg.render().asPng();
+	return buffer;
```

Comment:
Little worried we're rendering out a `png` from the SVG everytime the favicon is requested. I haven't done the perf testing yet, but isn't that a fairly heavy thing to do every time? Wondering if we should have some caching for this, or if there's a world where we return the SVG favicon for browsers that support svg favicons and have a default png that doesn't have to be generated each time as a fallback for browsers that don't 🤔

Reply:
Yep, using caching can definitely reduce the performance overhead. 

In additional, I will try detecting browser support via user agent to conditional handle it.